client/performance-profiler/pages/dashboard/index.tsx

```diff
@@ -71,6 +84,15 @@ export const PerformanceProfilerDashboard = ( props: PerformanceProfilerDashboar
 			? ( mobileReport as PerformanceReport )
 			: ( desktopReport as PerformanceReport );
 
+	if ( testStartTime.current && desktopLoaded && mobileLoaded ) {
+		recordTracksEvent( 'calypso_performance_profiler_test_completed', {
+			url: finalUrl,
```

Comment:
I have realised that for some reason I cannot see this `url` value in the track events payload (see screenshot below). It can be extracted from the `page` URL query parameter, but it's worth checking if it is being passed here.

![CleanShot 2024-09-10 at 17 52 06@2x](https://github.com/user-attachments/assets/6acd4f9b-feab-44a9-bff0-7688de76d54a)

Reply:
Yeah, I tested the changes again, and the URL prop only seems to be passed to the `calypso_performance_profiler_test_started` event correctly. I'll take a look why this happens and push a fix soon.

Reply:
Actually, I found that this is caused by the lack of the `finalUrl` field. After receiving the `hash` the basicMetricsQuery is invalidated, which sets the `finalUrl` variable to undefined.